terraform-aws-static-site/main.tf

```
@@ -0,0 +1,163 @@
+# Non www-bucket
+resource "aws_s3_bucket" "b" {
+  bucket = var.bucket_name
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+}
+
+data "aws_iam_policy_document" "b" {
+  statement {
+    sid    = "PublicReadGetObject"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
+  }
+}
+
+resource "aws_s3_bucket_policy" "b" {
+  bucket = aws_s3_bucket.b.id
+  policy = data.aws_iam_policy_document.b.json
+}
+
+# www-bucket
+resource "aws_s3_bucket" "www-bucket" {
+  bucket = "www-${var.bucket_name}"
+  acl    = "public-read"
+
+  website {
+    redirect_all_requests_to = "https://${var.redirect_address}"
+  }
+}
+
+# Non www-bucket CDN
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = aws_s3_bucket.b.bucket_regional_domain_name
+    origin_id   = var.origin_id
+
+    custom_origin_config {
+      origin_protocol_policy = "http-only"
+      http_port              = "80"
+      https_port             = "443"
+      origin_ssl_protocols   = ["TLSv1"]
+    }
+  }
+
+  price_class = "PriceClass_100"
+
+  enabled = true
+
+  aliases = var.aliases
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = var.origin_id
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
```

We also want to tweak this slightly, lets aggressively cache the page, and enable cloudfront compression to decrease our bundle size for browsers that can handle it.

```suggestion
    viewer_protocol_policy = "redirect-to-https"
    compress               = true
    min_ttl                = 0
    default_ttl            = 31557600
    max_ttl                = 31557600
```